### PR TITLE
enable user hooks only for internal backend

### LINKED ISSUE #91: deleting an LDAP user takes several minutes when jsxc.nextcloud is active
jsxc.nextcloud: 3.3.2

## Expected behaviour

When I delete a user from LDAP which isn't in LDAP anymore with `occ user:delete <user>` it should only take a couple of seconds.

## Actual behaviour

When jsxc.nextcloud is active, deleting such a user takes several minutes.
These are the SQL queries that the SQL server is receiving during that time:
```
SELECT "appid", "configkey", "configvalue" FROM "oc_preferences" WHERE "userid" = $1
```
```
SELECT "ldap_dn" FROM "oc_ldap_user_mapping" WHERE "owncloud_name" = $1
```
```
INSERT INTO "oc_ojsxc_stanzas" ("to", "from", "stanza") VALUES($1,$2,$3)
```

It looks like although I am using an external XMPP server, JSXC is trying to [keep the rosters for its internal server up to date](https://github.com/nextcloud/jsxc.nextcloud/blob/a4fb2f87bff2d18f88838b3519d9d061215d1e2e/build/lib/rosterpush.php#L77-L91) after a [user removal](https://github.com/nextcloud/jsxc.nextcloud/blob/a4fb2f87bff2d18f88838b3519d9d061215d1e2e/build/lib/hooks.php#L79-L88).

As a result it took up to 9 minutes to delete a user from (an admittedly large) LDAP.

### PR DESCRIPTION
**Pull request by sualko**
fix #91

### CODE REVIEW

File: lib/ContactsMenu/Providers/ChatProvider.php [hunk 1]
```diff
@@ -52,7 +53,7 @@ public function process(IEntry $entry)
 			$config = \OC::$server->getConfig();
 			$xmppPreferMail = $config->getAppValue('ojsxc', 'xmppPreferMail', false);
 			$xmppPreferMail = $xmppPreferMail === true || $xmppPreferMail === 'true';
-			$serverType = $config->getAppValue('ojsxc', 'serverType', 'external');
```
Comment (LEDfan) [inline lib/ContactsMenu/Providers/ChatProvider.php]: @sualko This was not intentional I guess? (That the default is external)

**Comment by LEDfan** [comment]
@sualko AFAIK the default the backend is internal, but this wasn't used in the check. I added a static method to ensure it's the same in the whole code base.

File: lib/ContactsMenu/Providers/ChatProvider.php (hunk 1 above)
Comment (sualko) [reply lib/ContactsMenu/Providers/ChatProvider.php]: You are right :+1:

Comment (sualko) [reply lib/ContactsMenu/Providers/ChatProvider.php]: btw I added another commit which substitutes all server type string literals with constants.

**Comment by LEDfan** [comment]
@sualko I just fixed an incompatibility with NC 14 (https://travis-ci.org/nextcloud/jsxc.nextcloud/jobs/350202403#L570)

**Comment by sualko** [comment]
This was already fixed in https://github.com/nextcloud/jsxc.nextcloud/pull/96

**Comment by sualko** [comment]
I will remove your commit and merge master with a forced push.

**Comment by LEDfan** [comment]
.... I just discovered that, removed my commit again.. Sorry :see_no_evil:

**Pull request merged**